tony-core/src/main/java/com/linkedin/tony/util/Utils.java

```diff
@@ -489,9 +500,33 @@ public static void createDir(FileSystem fs, Path dir, FsPermission permission) {
   }
 
   public static boolean isJobTypeTracked(String taskName, Configuration tonyConf) {
-    String[] ignoredJobTypes = tonyConf.getStrings(TonyConfigurationKeys.UNTRACKED_JOBTYPES, TonyConfigurationKeys.UNTRACKED_JOBTYPES_DEFAULT);
+    String[] ignoredJobTypes = tonyConf.getStrings(TonyConfigurationKeys.UNTRACKED_JOBTYPES,
+            TonyConfigurationKeys.UNTRACKED_JOBTYPES_DEFAULT);
     return !Arrays.asList(ignoredJobTypes).contains(taskName);
   }
 
+  public static void uploadFileAndSetConfResources(Path hdfsPath, Path filePath, String fileName,
+                                                   Configuration tonyConf, FileSystem fs,
+                                                   LocalResourceType resourceType, String resourceKey) throws IOException {
+    Path dst = new Path(hdfsPath, fileName);
+    HdfsUtils.copySrcToDest(filePath, dst, tonyConf);
+    fs.setPermission(dst, new FsPermission((short) 0770));
+    String dstAddress = dst.toString();
+    if (resourceType == LocalResourceType.ARCHIVE) {
+      dstAddress += Constants.ARCHIVE_SUFFIX;
+    }
+    appendConfResources(resourceKey, dstAddress, tonyConf);
+  }
+
+  public static void appendConfResources(String key, String resource, Configuration tonyConf) {
+    String[] resources = tonyConf.getStrings(key);
+    List<String> updatedResources = new ArrayList<>();
+    if (resources != null) {
+      updatedResources = new ArrayList<>(Arrays.asList(resources));
```

Comment:
I think the `new ArrayList<>()` wrapping is unneeded.

Reply:
It is needed because Array.asList returns an immutable list.